geoportailv3/static/js/search/searchdirective.js

```diff
@@ -0,0 +1,335 @@
+goog.require('ngeo.searchDirective');
+
+
+/**
+ * @return {angular.Directive} The Directive Object Definition
+ * @param {string} appSearchTemplateUrl
+ * @ngInject
+ */
+app.searchDirective = function(appSearchTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appSearchMap'
+    },
+    controller: 'AppSearchController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appSearchTemplateUrl,
+    link:
+        /**
+         * @param {angular.Scope} scope Scope
+         * @param {angular.JQLite} element Element
+         * @param {angular.Attributes} attrs Atttributes
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+
+app.module.directive('appSearch', app.searchDirective);
+
+
+
+/**
+ * @ngInject
+ * @constructor
+ * @param {angular.Scope} $scope Angular root scope.
+ * @param {app.Themes} appThemes Themes service.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ *     create GeoJSON Bloodhound service.o
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog
+ * @param {app.GetLayerForCatalogNode} appGetLayerForCatalogNode
+ * @param {app.ShowLayerinfo} appShowLayerinfo
+ * @param {string} searchServiceUrl
+ * @export
+ */
+app.SearchDirectiveController =
+    function($scope, appThemes, $compile,
+        ngeoCreateGeoJSONBloodhound, gettextCatalog,
+        appGetLayerForCatalogNode, appShowLayerinfo, searchServiceUrl) {
+  /**
+   * @type {Array}
+   */
+  this.layers = [];
+
+  /**
+   * @type {ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = this.createFeatureOverlay_();
+
+  /**
+   * @type {app.GetLayerForCatalogNode}
+   * @private
+   */
+  this.getLayerFunc_ = appGetLayerForCatalogNode;
+
+  /**
+   * @type {app.ShowLayerinfo}
+   * @private
+   */
+  this.showLayerinfo_ = appShowLayerinfo;
+
+  /** @type {Bloodhound} */
+  var POIBloodhoundEngine = this.createAndInitPOIBloodhound_(
+      ngeoCreateGeoJSONBloodhound, searchServiceUrl);
+  /** @type {Bloodhound} */
+  var LayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
+
+  $scope.$on('gettextLanguageChanged', goog.bind(function(event, args) {
+    this.createLocalAllLayerData_(
+        appThemes, LayerBloodhoundEngine, gettextCatalog);
+  }, this));
+
+
+  /** @type {TypeaheadOptions} */
+  this['options'] = {
+    highlight: true
+  };
+
+  /** @type {Array.<TypeaheadDataset>} */
+  this['datasets'] = [{
+    source: POIBloodhoundEngine.ttAdapter(),
+    displayKey: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: {
+      header: function() {
+        return '<div class="header" translate>Addresses</div>';
```

Comment:
@elemoine I meant in this case (line 125) , where we want to translate the Addresses label , I dont think this will get picked up by the script ?